sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/PropertiesUtils.java

```diff
@@ -114,4 +138,117 @@ public static void loadPropertiesFromInputStream(InputStream in, Properties prop
 		}
 	}
 
+	/**
+	 * Retrieves frontend properties from the specified URL that points to a SPA configuration file.
+	 *
+	 * @param url The URL to retrieve the frontend properties from.
+	 * @return A Properties object containing the retrieved frontend properties.
+	 */
+	public static Properties getFrontendPropertiesFromSpaConfigUrl(String url) {
+		Properties properties = new Properties();
+		try {
+			HttpClient httpClient = HttpClientBuilder.create().build();
+			HttpGet request = new HttpGet(url);
+			HttpResponse response = httpClient.execute(request);
+			HttpEntity entity = response.getEntity();
+			try (InputStream inputStream = entity.getContent()) {
+				String jsonData = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+				ObjectMapper objectMapper = new ObjectMapper();
+				JsonNode jsonNode = objectMapper.readTree(jsonData);
+				JsonNode frontendModules = jsonNode.get("frontendModules");
+				Iterator<Map.Entry<String, JsonNode>> modulesIterator = frontendModules.fields();
+				while (modulesIterator.hasNext()) {
+					Map.Entry<String, JsonNode> moduleEntry = modulesIterator.next();
+					String moduleName = moduleEntry.getKey();
+					String moduleVersion = moduleEntry.getValue().asText();
+					properties.setProperty("spa.frontendModules." + moduleName, moduleVersion);
+				}
+			}
+			catch (IOException e) {
+				log.error(e.getMessage(), e);
+			}
+		}
+		catch (IOException e) {
+			log.error(e.getMessage(), e);
+		}
+		return properties;
+	}
+
+	/**
+	 * Generates configuration properties based on the specified Artifact.
+	 *
+	 * @param artifact The Artifact object containing information about the configuration.
+	 * @return A Properties object containing the generated configuration properties.
+	 */
+	public static Properties getConfigurationProperties(Artifact artifact) {
+		Properties properties = new Properties();
+		properties.setProperty("config." + artifact.getArtifactId(), artifact.getVersion());
+		return properties;
+	}
+
+	/**
+	 * Retrieves backend properties from the specified ZIP file.
```

Comment:
What do you mean by backend properties? Do you mean backend database, tomcat, memory, disk space? 😊

Reply:
It should be distro properties.